src/main/java/io/neonbee/data/internal/metrics/ConfiguredDataVerticleMetrics.java

```diff
@@ -0,0 +1,156 @@
+        this.reportTimingMetric = reportTimingMetric;
+    }
+
+    /**
+     * Configure the {@link DataVerticleMetrics} to report.
+     *
+     * To enable all metrics, you must specify the enabled key with the value ture. Otherwise, no metrics will be
+     * reported. You can select the registry to use by specifying the meterRegistryName key with the name of the
+     * registry associated with the registry in the micrometer options.
+     *
+     * If you specify any of the configuration values "reportNumberOfRequests", "reportActiveRequests",
+     * "reportStatusCounter", "reportTiming", only the values configured as true will be reported. If you do not specify
+     * any of these values, all metrics are reported.
+     *
+     * Full example:
+     *
+     * <pre>
+     * {@code
+     * {
+     *     "enabled" : true,
+     *     "meterRegistryName" : "default",
+     *     "reportNumberOfRequests" : true,
+     *     "reportActiveRequests" : true
+     *     "reportStatusCounter" : true,
+     *     "reportTiming" : true
+     * }
+     * }
+     * </pre>
+     *
+     * @param metricsConfig {@link JsonObject} containing the metrics configuration.
+     * @return configured {@link DataVerticleMetrics}
+     */
+    public static DataVerticleMetrics configureMetricsReporting(JsonObject metricsConfig) {
+        if (metricsConfig != null && Boolean.TRUE.equals(metricsConfig.getBoolean(ENABLED))) {
+
+            String meterRegistryName =
+                    metricsConfig.getString(METER_REGISTRY_NAME, MicrometerMetricsOptions.DEFAULT_REGISTRY_NAME);
+            MeterRegistry registry = BackendRegistries.getNow(meterRegistryName);
+
+            if (registry == null) {
+                LOGGER.error(
+                        "Micrometer registry hasn't been registered yet or it has been stopped. Metrics will not be sent.");
+                return DUMMY_IMPL;
+            } else {
+                DataVerticleMetrics metricsImpl = new DataVerticleMetricsImpl(registry);
+
+                int fieldNameSize = metricsConfig.containsKey(METER_REGISTRY_NAME) ? 2 : 1;
+                boolean activateAllMetrics = metricsConfig.size() == fieldNameSize;
+                if (activateAllMetrics) {
+                    return metricsImpl;
+                } else {
+                    return new ConfiguredDataVerticleMetrics(
+                            Boolean.TRUE.equals(metricsConfig.getBoolean(NUMBER_OF_REQUESTS)) ? metricsImpl
+                                    : DUMMY_IMPL,
+                            Boolean.TRUE.equals(metricsConfig.getBoolean(ACTIVE_REQUESTS)) ? metricsImpl : DUMMY_IMPL,
+                            Boolean.TRUE.equals(metricsConfig.getBoolean(STATUS_COUNTER)) ? metricsImpl : DUMMY_IMPL,
+                            Boolean.TRUE.equals(metricsConfig.getBoolean(TIMING)) ? metricsImpl : DUMMY_IMPL);
+                }
+            }
+        } else {
+            return DUMMY_IMPL;
+        }
```

Comment:
nit; I think it would be nicer to read if we would do an early return here, and get rid of one level of nested `if`'s.

```suggestion
        if (metricsConfig == null && Boolean.TRUE.equals(metricsConfig.getBoolean(ENABLED))) {
            return DUMMY_IMPL;
        }

            String meterRegistryName =
                    metricsConfig.getString(METER_REGISTRY_NAME, MicrometerMetricsOptions.DEFAULT_REGISTRY_NAME);
            MeterRegistry registry = BackendRegistries.getNow(meterRegistryName);

            if (registry == null) {
                LOGGER.error(
                        "Micrometer registry hasn't been registered yet or it has been stopped. Metrics will not be sent.");
                return DUMMY_IMPL;
            } else {
                DataVerticleMetrics metricsImpl = new DataVerticleMetricsImpl(registry);

                int fieldNameSize = metricsConfig.containsKey(METER_REGISTRY_NAME) ? 2 : 1;
                boolean activateAllMetrics = metricsConfig.size() == fieldNameSize;
                if (activateAllMetrics) {
                    return metricsImpl;
                }
                return new ConfiguredDataVerticleMetrics(
                        Boolean.TRUE.equals(metricsConfig.getBoolean(NUMBER_OF_REQUESTS)) ? metricsImpl : DUMMY_IMPL,
                        Boolean.TRUE.equals(metricsConfig.getBoolean(ACTIVE_REQUESTS)) ? metricsImpl : DUMMY_IMPL,
                        Boolean.TRUE.equals(metricsConfig.getBoolean(STATUS_COUNTER)) ? metricsImpl : DUMMY_IMPL,
                        Boolean.TRUE.equals(metricsConfig.getBoolean(TIMING)) ? metricsImpl : DUMMY_IMPL);
            }
        }
```

Reply:
I agree, it is nicer to read, but I disagree with your suggested code. You forgot to invert the Boolean.TRUE.equals(metricsConfig.getBoolean(ENABLED)) part.


```
if (metricsConfig == null || Boolean.FALSE.equals(metricsConfig.getBoolean(ENABLED, Boolean.FALSE))) {
    return DUMMY_IMPL;
}
```

Reply:
My suggestion is also wrong. It has to be like this:

```
if (metricsConfig == null || !Boolean.TRUE.equals(metricsConfig.getBoolean(ENABLED))) {
    return DUMMY_IMPL;
}
```